eclass/multilib.eclass

```diff
@@ -425,29 +428,32 @@ multilib_env() {
 			export CFLAGS_ilp32d=${CFLAGS_ilp32d--mabi=ilp32d -march=rv32imafdc}
 			export CHOST_ilp32d=${CTARGET/riscv64/riscv32}
 			export CTARGET_ilp32d=${CTARGET/riscv64/riscv32}
-			export LIBDIR_ilp32d=${LIBDIR_ilp32d-lib32/ilp32d}
+			export LIBDIR_ilp32d=${LIBDIR_ilp32d-lib/ilp32d}
 
 			export CFLAGS_ilp32=${CFLAGS_ilp32--mabi=ilp32 -march=rv32imac}
 			export CHOST_ilp32=${CTARGET/riscv64/riscv32}
 			export CTARGET_ilp32=${CTARGET/riscv64/riscv32}
-			export LIBDIR_ilp32=${LIBDIR_ilp32-lib32/ilp32}
+			export LIBDIR_ilp32=${LIBDIR_ilp32-lib/ilp32}
 
-			: ${MULTILIB_ABIS=lp64d lp64 ilp32d ilp32}
-			: ${DEFAULT_ABI=lp64d}
+			local __libdir_riscvdefaultabi_variable="LIBDIR_${DEFAULT_ABI}"
+			export ${!__libdir_riscvdefaultabi_variable}="lib64"
```

Comment:
This does not work with `!` here (too many dereferences):
`export ${!__libdir_riscvdefaultabi_variable}="lib64"` is expanded to `export lib64/lp64d=lib64`
`export ${__libdir_riscvdefaultabi_variable}="lib64"` is expanded to `export LIBDIR_lp64d=lib64`

However there is no need for that intermediate variable `__libdir_riscvdefaultabi_variable`. Single-statement solution is any of these:
```
eval "export LIBDIR_${DEFAULT_ABI}=\"lib64\""
eval "export LIBDIR_${DEFAULT_ABI}=lib64"
```

Reply:
Surprising, but true. And fixed now.

That wasn't the only problem with the eclass though; the logic in case the libdir was already assigned elsewhere was wrong so far. (Never mind how likely the case of a *different* assignment elsewhere is...) Should also be fixed now.